### ISSUE TITLE
Incorrect `array-bracket-spacing`-error when using Flow type annotation

### ISSUE THREAD

**Issue by ahstro** [eslint/eslint#7468]
ESLint throws an `array-bracket-spacing`-error because of the space between my array brackets and the Flow type annotation.

**Tell us about your environment**

* **ESLint Version:** 3.8.1
* **Node Version:** 6.4.0
* **npm Version:** 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**

```
{
  "parser": "babel-eslint",
  "plugins": [
      "flowtype",
      "import",
      "react"
  ],
  "extends": [
    "eslint:recommended",
  ],
  "rules": {
    "comma-dangle": 0,
    "no-unexpected-multiline": 2,
    "accessor-pairs": 2,
    "consistent-return": 2,
    "curly": [2, "all"],
    "dot-location": [2, "property"],
    "dot-notation": 2,
    "eqeqeq": [2, "smart"],
    "no-console": 2,
    "no-caller": 2,
    "no-extra-bind": 2,
    "no-floating-decimal": 2,
    "no-labels": [2, {"allowLoop": true, "allowSwitch": true}],
    "no-lone-blocks": 2,
    "no-multi-spaces": 2,
    "no-new-func": 2,
    "no-octal-escape": 2,
    "no-param-reassign": 2,
    "no-proto": 2,
    "no-redeclare": 2,
    "no-return-assign": 2,
    "no-script-url": 2,
    "no-self-compare": 2,
    "no-sequences": 2,
    "no-unused-expressions": 2,
    "no-unused-vars": [2, { "varsIgnorePattern": "^_" , "argsIgnorePattern": "^_" }],
    "no-warning-comments": [2, { "terms": ["STOP"] }],
    "no-with": 2,
    "no-shadow-restricted-names": 2,
    "array-bracket-spacing": [2, "never"],
    "block-spacing": 2,
    "brace-style": 2,
    "camelcase": [2, {"properties": "never"}],
    "comma-spacing": 2,
    "comma-style": 2,
    "computed-property-spacing": 2,
    "eol-last": 2,
    "func-style": [2, "declaration", { "allowArrowFunctions": true }],
    "indent": [2, 2, {"VariableDeclarator": {"var": 2, "let": 2, "const": 3}}],
    "jsx-quotes": 2,
    "keyword-spacing": [2, {"before": true, "after": true, "overrides": {}}],
    "key-spacing": 2,
    "linebreak-style": 2,
    "no-lonely-if": 2,
    "no-multiple-empty-lines": [2, {"max": 2}],
    "no-spaced-func": 2,
    "no-trailing-spaces": 2,
    "operator-linebreak": 2,
    "quote-props": [2, "as-needed"],
    "quotes": [2, "double", "avoid-escape"],
    "semi-spacing": 2,
    "semi": [2, "never"],
    "space-before-blocks": 2,
    "space-before-function-paren": [2, "never"],
    "space-in-parens": 2,
    "space-infix-ops": 2,
    "space-unary-ops": 2,
    "arrow-parens": 0,
    "arrow-spacing": 2,
    "generator-star-spacing": 2,
    "no-const-assign": 2,
    "no-var": 2,
    "one-var": [2, "never"],
    "object-shorthand": 2,
    "prefer-arrow-callback": 2,
    "prefer-const": 2,
    "prefer-template": 2,
    "require-yield": 2,
    "react/jsx-boolean-value": [2, "never"],
    "react/jsx-curly-spacing": 2,
    "react/jsx-no-bind": 2,
    "react/jsx-no-duplicate-props": 2,
    "react/jsx-no-undef": 2,
    "react/jsx-uses-vars": 2,
    "react/jsx-uses-react": 2,
    "react/jsx-wrap-multilines": 2,
    "react/no-direct-mutation-state": 2,
    "react/no-unknown-property": 2,
    "react/prop-types": 2,
    "react/react-in-jsx-scope": 2,
    "import/extensions": [2, {
      "json": "always",
      "css": "always",
      "js": "never"
    }],
    "flowtype/boolean-style": [2, "boolean"],
    "flowtype/define-flow-type": 1,
    "flowtype/delimiter-dangle": [2, "always-multiline"],
    "flowtype/generic-spacing": [2, "never"],
    "flowtype/no-weak-types": 2,
    "flowtype/object-type-delimiter": [2, "comma"],
    "flowtype/require-parameter-type": [2, {
      "excludeParameterMatch": "^_"
    }],
    "flowtype/require-return-type": [2, "always", {
      "annotateUndefined": "never"
    }],
    "flowtype/require-valid-file-annotation": 2,
    "flowtype/semi": [2, "never"],
    "flowtype/space-after-type-colon": [2, "always"],
    "flowtype/space-before-generic-bracket": [2, "never"],
    "flowtype/space-before-type-colon": [2, "never"],
    "flowtype/type-id-match": [2, "^([A-Z][A-Za-z0-9]+)+$"],
    "flowtype/union-intersection-spacing": [2, "always"],
    "flowtype/use-flow-type": 1,
  },
  "env": {
    "browser": true,
    "mocha": true,
    "es6": true,
    "node": true
  },
  "parserOptions": {
    "ecmaVersion": 6,
    "ecmaFeatures": {
      "jsx": true,
      "experimentalObjectRestSpread": true
    },
    "sourceType": "module",
  },
  "settings": {
    "flowtype": {
      "onlyFilesWithFlowAnnotation": true
    }
  },
  "globals": {
    "chrome": true,
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
function refreshToHighlights([e, words]: Refresh): Rx.Observable {
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**

```
   78:42  error  There should be no space before 'Refresh'                     array-bracket-spacing
```

**Comment by platinumazure** [comment]
Hi @ahstro, thanks for the issue!

This should be fixed by #7445, which is not yet merged but which we might merge and release today if all goes well. Please follow that PR. I'm going to close this for now, but please reopen if #7445 does not solve your issue.
